language/move-vm/runtime/src/interpreter.rs

```diff
@@ -65,6 +65,10 @@ pub(crate) struct Interpreter<L: LogContext> {
     call_stack: CallStack,
     // Logger to report information to clients
     log_context: L,
+
+    // Scrach space to prevent allocations
```

Comment:
I don't buy this. At all. I understand the intent but there are a ton of different ways we may want to tackle this issue and for now a scratch space is the last thing I want ;)
having said so I'd be surprised if this leads to any perf win at all. But I could be wrong and curious to see that.
And if it did maybe it would be a reason to go for a review of our arguments passing for natives

Reply:
Yes, I agree this memory reuse is a little on the strange side. In theory an allocator should be doing exactly this. I will try to isolate this change, and measure it. However I did got there because the interpreter did show up on the profiler as making a whole lot of alloc/dealloc calls.

But I agree this is not a clean solution in any case.